File: ios_src/Shuiguang2/Shuiguang/AuthsTask.m
//
//  AuthsTask.m
//  Shuiguang
//
//  Created by dehualai on 3/1/17.
//  Copyright © 2017 NanoMed. All rights reserved.
//

#import "AuthsTask.h"
#import <BmobSDK/BmobUser.h>

@implementation AuthsTask


// Override 
-(void)sync:(NSDictionary*)params{

    NSNumber* task = (NSNumber*) [params objectForKey:PARAM_TASK];
    switch (task.intValue) {
        case TASK_REGISTER:
            [self authRegister:params];
            break;
        case TASK_LOGIN:
            [self authLogin:params];
            break;
        default:
            break;
    }
}



-(void)authRegister:(NSDictionary*)params{
    NSNumber* type = (NSNumber*) [params objectForKey:PARAM_AUTH_TYPE];
    switch (type.intValue) {
        case TYPE_PWD:
        {
            NSString* username = (NSString*) [params objectForKey:PARAM_USERNAME];
            NSString* email = (NSString*) [params objectForKey:PARAM_EMAIL];
            NSString* password = (NSString*) [params objectForKey:PARAM_PASSWORD];
            username = [[username stringByReplacingOccurrencesOfString:@" " withString:@""] lowercaseString];
            email = [[email stringByReplacingOccurrencesOfString:@" " withString:@""] lowercaseString];
            password = [password stringByReplacingOccurrencesOfString:@" " withString:@""];
            BmobUser *bUser = [[BmobUser alloc] init];
            [bUser setUsername:username];
            [bUser setEmail:email];
            [bUser setPassword:password];
            [bUser signUpInBackgroundWithBlock:^ (BOOL isSuccessful, NSError *error){
                if (isSuccessful){
                    NSLog(@"Sign up successfully");
                    if (self.handler) [self.handler onSuccess:bUser];
                } else {
                    NSLog(@"%@",error);
                    if (self.handler) [self.handler onFailure:error.domain errorCode:@(error.code)];
                }
            }];
        }
        break;
        case TYPE_WECHAT:
        {
        
        }
        break;
        default:
            break;
    }

}


-(void)authLogin:(NSDictionary*)params{
    NSNumber* type = (NSNumber*) [params objectForKey:PARAM_AUTH_TYPE];
    switch (type.intValue) {
        case TYPE_PWD:
        {
            NSString* username = (NSString*) [params objectForKey:PARAM_USERNAME];
            NSString* password = (NSString*) [params objectForKey:PARAM_PASSWORD];
            username = [[username stringByReplacingOccurrencesOfString:@" " withString:@""] lowercaseString];
            password = [password stringByReplacingOccurrencesOfString:@" " withString:@""];
            [BmobUser loginInbackgroundWithAccount:username andPassword:password block:^(BmobUser *user, NSError *error) {
                if (user) {
                    NSLog(@"SUCCESS %@",user);
                    if (self.handler) [self.handler onSuccess:user];
                } else {
                    NSLog(@"FAILURE %@",error);
                    if (self.handler) [self.handler onFailure:error.domain errorCode:@(error.code)];
                }
            }];
        }
        break;
        case TYPE_WECHAT:
        {
        
        }
        break;
        default:
            break;
    }
}


@end

File: android_v2/app/src/main/java/com/mlz/shuiguang/transmission/task/AuthsTask.java
package com.mlz.shuiguang.transmission.task;

import android.content.Context;
import android.util.Log;

import com.mlz.shuiguang.transmission.SyncTask;
import com.mlz.shuiguang.transmission.bmobObject.User;
import com.mlz.shuiguang.util.Global;
import com.mlz.shuiguang.util.Util;

import java.util.HashMap;

import cn.bmob.v3.exception.BmobException;
import cn.bmob.v3.listener.SaveListener;

/**
 * Created by dehualai on 1/13/17.
 */

public class AuthsTask extends SyncTask {

    public static final int TASK_REGISTER = 0;
    public static final int TASK_LOGIN = 1;
    public static final String PARAM_AUTH_TYPE = "auth_type";
    public static final int TYPE_PWD = 0;
    public static final int TYPE_WECHAT = 1;
    public static final String PARAM_EMAIL = "email";
    public static final String PARAM_USERNAME = "username";
    public static final String PARAM_PASSWORD = "password";

    public AuthsTask(Context context){
        super(context);
    }

    @Override
    public void sync(HashMap<Object, Object> params) {
        Integer task = (Integer) params.get(PARAM_TASK);
        switch (task){
            case TASK_REGISTER:
                register(params);
                break;
            case TASK_LOGIN:
                login(params);
                break;
            default:
                break;
        }
    }

    private void register(HashMap<Object, Object> params){
        Integer type = (Integer) params.get(PARAM_AUTH_TYPE);
        switch (type){
            case TYPE_PWD:

                /*
                String username = (String)params.get(PARAM_USERNAME);
                String password = (String)params.get(PARAM_PASSWORD);
                String email = (String)params.get(PARAM_EMAIL);
                User user = new User();
                user.setUsername(username.replace(" ", "").toLowerCase());
                user.setPassword(password);
                user.setEmail(email.replace(" ", "").toLowerCase());
                user.signUp(new SaveListener<User>(){
                    @Override
                    public void done(User u, BmobException e) {
                        if(e==null){
                            Log.d("show", "添加数据成功，返回objectId为："+u.getObjectId());
                            if (handler!=null) handler.onSuccess(u);
                        }else{
                            Log.d("show", "创建数据失败：" + e.getMessage() + " code: " + e.getErrorCode());
                            if (handler!=null) handler.onFailure(e.getMessage(), e.getErrorCode());
                        }
                    }
                });

                */
                break;
            case TYPE_WECHAT:
                break;
            default:
                break;
        }
    }

    private void login(HashMap<Object, Object> params){
        Integer type = (Integer) params.get(PARAM_AUTH_TYPE);
        switch (type){
            case TYPE_PWD:

                /*
                String username = (String)params.get(PARAM_USERNAME);
                String password = (String)params.get(PARAM_PASSWORD);
                User user = new User();
                user.setUsername(username.replace(" ", "").toLowerCase());
                user.setPassword(password);
                user.login(new SaveListener<User>(){
                    @Override
                    public void done(User u, BmobException e) {
                        if(e==null){
                            Log.d("show", "登录成功，返回objectId为："+u.getObjectId());
                            if (handler!=null) handler.onSuccess(u);
                        }else{
                            Log.d("show", "登录失败：" + e.getMessage() + " code: " + e.getErrorCode());
                            if (handler!=null) handler.onFailure(e.getMessage(), e.getErrorCode());
                        }
                    }
                });

*/

                break;
            default:
                break;
        }
    }
}
